grades/models.py

```diff
@@ -79,3 +82,49 @@ def __str__(self):
             user=self.final_grade.user,
             course_id=self.final_grade.course_run.edx_course_key
         )
+
+
+class CourseRunGradingStatus(TimestampedModel):
+    """
+    Additional information for the course run related to the final grades
+    """
+    course_run = models.OneToOneField(CourseRun, null=False)
+    status = models.CharField(
+        null=False,
+        choices=[(status, status) for status in FinalGradeStatus.ALL_STATUSES],
+        default=FinalGradeStatus.PENDING,
+        max_length=30,
+    )
+
+    @classmethod
+    def is_complete(cls, course_run):
+        """
+        Returns True if there is an entry with status 'complete'
+        """
+        return cls.objects.filter(course_run=course_run, status=FinalGradeStatus.COMPLETE).exists()
+
+    @classmethod
+    def is_pending(cls, course_run):
+        """
+        Returns True if there is an entry with status 'pending'
+        """
+        return cls.objects.filter(course_run=course_run, status=FinalGradeStatus.PENDING).exists()
+
+    @classmethod
+    def set_to_complete(cls, course_run):
+        """
+        Sets the status for the course_run to complete
+        """
+        course_run_grade_status, _ = cls.objects.get_or_create(course_run=course_run)
+        if course_run_grade_status.status != FinalGradeStatus.COMPLETE:
```

Comment:
It looks like this line is only partially covered. Can you add a test that asserts that nothing happens if all grades are already frozen?